src/components/ButtonWithIcon/ButtonWithIcon.jsx

```diff
@@ -0,0 +1,110 @@
+import _ from 'lodash';
+import React from 'react';
+import ReactDOM from 'react-dom';
+import PropTypes from 'react-peek/prop-types';
+import Button from '../Button/Button';
+import Icon from '../Icon/Icon';
+import { lucidClassNames } from '../../util/style-helpers.js';
+import { createClass } from '../../util/component-types';
+
+const cx = lucidClassNames.bind('&-ButtonWithIcon');
+
+const { element, string, oneOfType, node, arrayOf } = PropTypes;
+
+const ButtonWithIcon = createClass({
+	displayName: 'ButtonWithIcon',
+
+	statics: {
+		peek: {
+			description: `
+				A basic button with an \`Icon\`.
+			`,
+			categories: ['controls', 'buttons'],
+			madeFrom: ['Button', 'Icon'],
+		},
+	},
+
+	propName: 'ButtonWithIcon',
+
+	propTypes: {
+		...Button.propTypes,
+		...Icon.propTypes,
+
+		className: string`
+			Class names that are appended to the defaults
+		`,
+
+		children: oneOfType([node, arrayOf(node)])`
+			Any valid React children
+		`,
+
+		type: string`
+			Form element type variations of Button. Passed through to DOM Button.
+		`,
+
+		icon: element`
+			The \`Icon\` displayed inside the Button.
+		`,
+	},
+
+	getDefaultProps() {
+		return {
+			isDisabled: false,
+			isActive: false,
+			onClick: _.noop,
+			type: 'button',
+		};
+	},
+
+	handleClick(event) {
+		const { isDisabled, onClick } = this.props;
+		const domNode = ReactDOM.findDOMNode(this);
```

Comment:
instead of finding a dom node everytime the buutton is clicked, you should store the `ref` of the thing you want to focus. See https://reactjs.org/docs/refs-and-the-dom.html

Also, right now, this focuses the `div`, is that intended? You probably want to store the ref of the `Button` so that the button is focused instead.